cmd/s2i/main.go

```diff
@@ -171,6 +171,7 @@ $ s2i build . centos/ruby-22-centos7 hello-world-app
 	buildCmd.Flags().StringVarP(&(cfg.Ref), "ref", "r", "", "Specify a ref to check-out")
 	buildCmd.Flags().StringVarP(&(cfg.AssembleUser), "assemble-user", "", "", "Specify the user to run assemble with")
 	buildCmd.Flags().StringVarP(&(cfg.ContextDir), "context-dir", "", "", "Specify the sub-directory inside the repository with the application sources")
+	buildCmd.Flags().StringVarP(&(cfg.ExcludeRegExp), "exclude", "", "", "Regular expression for selecting files from the source tree to exclude from the tar stream; see https://golang.org/pkg/regexp for syntax")
```

Comment:
hey @bparees @mfojtik - was reading the exchange at the doctor's this AM, but decided not to chime in until I got in front of the computer (that also allowed me to think a bit about it).
1) with the --*-git version, I can't help but circle back to the "what happens when we support other SCMs", and yes there is the "how likely is that" rebuttal, but my gut tells me it is better to be generic here ... though it adds more to-do's (see 2) and 3) )
2) that said, I think the "" use case isn't at the way we want it (since not specifying a string is "" today with golang) ... I believe I need to rework the impl so the actual default regexp is the default value for --exclude, and I update the help to a) avoid the tar stream ref as @bparees noted, but b) add a bit to explain the default excludes .git, but still note the golang ref
3) will add test case to confirm the "" behavior ... if it matches everything, will need to special case it

And yes, at least for now, even with 2) and 3), I'm still of the mind to keep it generic, but understand my footing there may get too problematic.  I'll update in a bit with findings.